src/main.rs

```diff
@@ -78,27 +80,34 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // creating a 'service' to handle requests for that specific connection.
     let service = service::make_service_fn(move |addr: &AddrStream| {
         trace!("Connection from: {:?}", addr);
-        let client = client.clone();
+        let client_map = client_map.clone();
 
         #[cfg(feature = "expose-metrics")]
         let handle = handle.clone();
 
         async move {
             Ok::<_, RequestError>(service::service_fn(move |incoming: Request<Body>| {
+                let token = incoming
+                    .headers()
+                    .get("authorization")
+                    .map(|value| value.to_str().ok())
```

Comment:
Couldn't this be a `flat_map` instead of `map().flatten()`

Reply:
Yeah, `and_then` can be used here